File: Backend/scheduling/src/main/java/com/draveterinaria/cl/scheduling/model/SubtipoServicio.java
package com.draveterinaria.cl.scheduling.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "SUBTIPO_SERVICIO")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class SubtipoServicio {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "subtipo_seq")
    @SequenceGenerator(name = "subtipo_seq", sequenceName = "SUBTIPO_SEQ", allocationSize = 1)
    private Long idSubtipo;

    @Column(name = "NOMBRE_SUBTIPO")
    private String nombreSubtipo;

    @Column(name = "PRECIO")
    private Double precio;

    @ManyToOne
    @JoinColumn(name = "ID_TIPO_SERVICIO", nullable = false)
    @JsonBackReference
    private TipoServicio tipoServicio;
}

File: app/src/main/java/com/example/draveterinaria/data/model/SubtipoServicio.kt
package com.example.draveterinaria.data.model

import com.fasterxml.jackson.annotation.JsonBackReference
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.SequenceGenerator
import jakarta.persistence.Table


@Entity
@Table(name = "SUBTIPO_SERVICIO")

class SubtipoServicio {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "subtipo_seq")
    @SequenceGenerator(name = "subtipo_seq", sequenceName = "SUBTIPO_SEQ", allocationSize = 1)
     var idSubtipo: Long? = null

    @Column(name = "NOMBRE_SUBTIPO")
    var nombreSubtipo: String? = null

    @Column(name = "PRECIO")
    var precio: Double? = null

    @ManyToOne
    @JoinColumn(name = "ID_TIPO_SERVICIO", nullable = false)
    @JsonBackReference
    var tipoServicio: TipoServicio? = null
}